pkg/devserver/devserver.go

```diff
@@ -252,3 +267,84 @@ func createInmemoryRedis(ctx context.Context) (rueidis.Client, error) {
 	}()
 	return rc, nil
 }
+
+func getSendingEventHandler(ctx context.Context, pb pubsub.Publisher, topic string) execution.HandleSendingEvent {
+	return func(ctx context.Context, evt event.Event, item queue.Item) error {
+		byt, err := json.Marshal(evt)
+		if err != nil {
+			return fmt.Errorf("error marshalling invocation event: %w", err)
+		}
+
+		err = pb.Publish(
+			ctx,
+			topic,
+			pubsub.Message{
+				Name:      event.EventReceivedName,
+				Data:      string(byt),
+				Timestamp: time.Now(),
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("error publishing invocation event: %w", err)
+		}
+
+		return nil
+	}
+}
+
+func getInvokeNotFoundHandler(ctx context.Context, pb pubsub.Publisher, topic string) execution.InvokeNotFoundHandler {
+	return func(ctx context.Context, opts execution.InvokeNotFoundHandlerOpts) error {
+		now := time.Now()
+		data := map[string]interface{}{
+			"function_id": opts.FunctionID,
+			"run_id":      opts.RunID,
+		}
+
+		origEvt := opts.OriginalEvent.GetEvent().Map()
+		if dataMap, ok := origEvt["data"].(map[string]interface{}); ok {
+			if inngestObj, ok := dataMap[consts.InngestEventDataPrefix].(map[string]interface{}); ok {
+
+				if dataValue, ok := inngestObj[consts.InvokeCorrelationId].(string); ok {
+					logger.From(ctx).Debug().Str("data_value_str", dataValue).Msg("data_value")
+					data[consts.InvokeCorrelationId] = dataValue
+				}
+			}
+		}
+
+		if opts.Err != nil {
+			data["error"] = opts.Err
+		} else {
+			data["result"] = opts.Result
```

Comment:
This seems like it's repeated here and on our cloud.  Is there a way we can normalize this into one function which we use both here and imported into our cloud repo?

Reply:
I still see this logic - specifically from line 305 -> 325 - duplicated between here and the cloud.  Ideally we'd extract this into a function such that the actual _event generation itself_ is written once and imported.